### PR TITLE
Add new downloads table for the Bio-Formats C++ artifacts

### PR DESCRIPTION
**Pull request by sbesson**
This should set-up all the infrastructure required for exposing the Bio-Formats C++ artifacts in the 5.1.x series. /cc @rleigh-dundee @melissalinkert

I am currently using the generic OME icon but we may want to have dedicated icons for each binary /cc @gusferguson

--no-rebase

### CODE REVIEW

**Comment by gusferguson** [comment]
@sbesson - just let me know what you want - easy to do.

**Comment by ghost** [comment]
@gusferguson Discussing with @sbesson we would like a row of icons for each C++ build:

"zip" "bfcpp" "platform"

Where "zip" is the existing zip logo, "bfcpp" is the bioformats logo with "C++" overlaid like the matlab logo, and then "platform" is the logo of the operating system.  We can reuse the Windows and MacOS logos here, but for Linux we want the specific distributions rather than the generic penguin.  For example:
- http://upload.wikimedia.org/wikipedia/commons/3/3f/Fedora_logo.svg
- https://design.ubuntu.com/brand/ubuntu-logo (orange circle with white elements)
- http://wiki.centos.org/ArtWork/MarketingMaterial?action=AttachFile&do=get&target=centos-official.svg  (from http://wiki.centos.org/ArtWork/MarketingMaterial)

We might need others in the future; at present we will need CentOS with Ubuntu and Fedora for the next milestone.

Thanks,
Roger

**Comment by gusferguson** [comment]
@sbesson @rleigh-dundee - No problem - will take a little longer as I will have to hunt up suitably transparent logos for the UNIX logos. Just finishing off something else then will get to it. Should be done by mid-tomorrow at latest all being equal.

**Comment by ghost** [comment]
@gusferguson The logo pages/files linked above should be the original vector graphics and should be directly usable.

**Comment by gusferguson** [comment]
@rleigh-dundee - Great thanks - that is hugely helpful.

**Comment by ghost** [comment]
https://design.ubuntu.com/downloads?metadata=element-logo+brand-ubuntu is linked from the above page and has the logos.

**Comment by ghost** [comment]
@gusferguson No specific "C++" logo I'm afraid, but https://www.google.co.uk/search?q=c%2B%2B+logo&client=firefox-a&hs=aVI&rls=org.mozilla:en-GB:official&channel=sb&tbm=isch&tbo=u&source=univ&sa=X&ei=D8jsU9yxA4Kp7AaJj4H4CQ&sqi=2&ved=0CB8QsAQ&biw=1536&bih=849 might provide some ideas!  Just the text "C++" is probably sufficient.

**Comment by gusferguson** [comment]
@rleigh-dundee - for some of the others I just used text C++ and it looked OK - I will run a draft by you tomorrow.

**Comment by sbesson** [comment]
See https://downloads.openmicroscopy.org/bio-formats/5.1.0-m0.4/#cpp for the output of  7fc1708

**Comment by gusferguson** [comment]
@sbesson @rleigh-dundee - Sorry if I am being a bit slow on this, but please can you confirm the text for the buttons - at the moment they all seem to be "Source Code" - do you wish to stick with that with just the logos varying?

**Comment by ghost** [comment]
@gusferguson For the various Linux logos, these are builds and not source code so I don't think anything needs overlaying.

**Comment by gusferguson** [comment]
@rleigh-dundee - the question is what text you want on the buttons? - for example:

bf_package

**Comment by ghost** [comment]
@sbesson Might have a better idea here, but "Bioformats binary" or "Bioformats build" would be fine with me.  However, making a separate graphic button for every single platform-specific build isn't going to be ideal--it's a lot of unnecessary work and it will be changing as we add more and more builds.  The text description is really going to need the platform and verson e.g. "Bioformats C++ binaries for CentOS 6.5", "Bioformats C++ binaries for MacOS X 10.8", etc.

Can't we make the "button" directly in the HTML i.e. a box with the icons plus text?  This would make it much easier to maintain and update; the icons can just be split out as separate files and included as needed.  This will make updating the platform and version trivial.

**Comment by gusferguson** [comment]
@rleigh-dundee @sbesson 

Have fun doing that in Sphinx!!! 

It takes me about 1 minute per button to change version numbers and generate new buttons. Quite happy to do that as I do for other buttons.

**Comment by ghost** [comment]
Note that this page isn't using sphinx, it's a plain HTML template.

**Comment by gusferguson** [comment]
That should make life easier!
When you have decided what you need let me know - if you can specify each button in terms of logos and text - the e.g. above would be:

zip download logo - Bio-formats logo - ImageJ logo - OME logo - "Bioformats Package"

**Comment by sbesson** [comment]
Deferring to @rleigh-dundee for the text decision here. My preference between his two options is definitely to use `binary` over `build` if we decide to add some text to the button.

**Comment by ghost** [comment]
For the source, we're currently using the generic OME logo; could we make that use the bioformats logo instead?

For the C++ binaries, lets go with:
- zip download logo - bioformats logo with C++ - CentOS logo - "CentOS 6.5 C++ binaries"
- zip download logo - bioformats logo with C++ - MacOS logo - "MacOS X 10.8 C++ binaries"
- zip download logo - bioformats logo with C++ - MacOS logo - "MacOS X 10.9 C++ binaries"

and for future milestone use:
- zip download logo - bioformats logo with C++ - CentOS logo - "CentOS 7 C++ binaries"
- zip download logo - bioformats logo with C++ - CentOS logo - "CentOS 7.1 C++ binaries"
- zip download logo - bioformats logo with C++ - CentOS logo - "CentOS 7.2 C++ binaries"
- zip download logo - bioformats logo with C++ - MacOS logo - "MacOS X 10.10 C++ binaries"
- zip download logo - bioformats logo with C++ - Fedora logo - "Fedora 20 C++ binaries"
- zip download logo - bioformats logo with C++ - Fedora logo - "Fedora 21 C++ binaries"
- zip download logo - bioformats logo with C++ - Ubuntu logo - "Ubuntu 12.04 C++ binaries"
- zip download logo - bioformats logo with C++ - Ubuntu logo - "Ubuntu 14.04 C++ binaries"
- zip download logo - bioformats logo with C++ - Ubuntu logo - "Ubuntu 14.10 C++ binaries"

Thanks,
Roger

**Comment by ghost** [comment]
Note that the future use ones aren't needed immediately; let's just go with the top three for now and see how they look.

**Comment by sbesson** [comment]
First icon proposal of @gusferguson now live at https://downloads.openmicroscopy.org/bio-formats/5.1.0-m0.4/#cpp

**Comment by ghost** [comment]
@gusferguson I'm not too convinced of the legibility of the operating system icons here; could we have them separate?  i.e. "zip" "bfcpp" "platform" as three individual icons?

**Comment by ghost** [comment]
The "C++" overlay is fine, but does this need anything to make it stand out a little from the underlying bf icon?  e.g. an outline, drop shadow/highlight?

**Comment by hflynn** [comment]
I think they are a bit too cluttered, I would remove the C++ bits from the icons since it is already in the heading and the text part of the button. The icon bit would then be two images like the ones above. Either that or remove the operating system icons and have the BF logo with C++ over it - but the first approach would look more consistent with the other stuff on the page I think.

**Comment by sbesson** [comment]
See https://downloads.openmicroscopy.org/bio-formats/5.1.0-m0.4/#cpp for the Bio-Formats source code icon.

File: bf_downloads.html [hunk 1]
```diff
@@ -186,6 +187,69 @@
...
+                    </table>
+                </div>
+                <ul>
+                    <li>
+                        <p>Documentation on how to build Bio-Formats C++ from source  is available to read on the web <a
+                                href="@DOC_URL@/developers/buildBF-CPP.html">here</a>.</p>
+                    </li>
+                    <li>
+                        <p>The Bio-Formats C++ Doxygen documentation is available
+                            to read on the web <a href="doxygen/">here</a></p>
```
Comment (hflynn) [inline bf_downloads.html]: This needs a full stop too so it matches the sentence above.

**Comment by ghost** [comment]
It's looking pretty good now.  The only remaining discrepancy is the source code download button width vs. the source code links width.  Could we either use the old button width in this section or expand the link button widths to match?  Maybe use a consistent size for all of them on the entire page?

**Comment by sbesson** [comment]
https://downloads.openmicroscopy.org/bio-formats/5.1.0-m0.4/ is now updated with the long versions of the icons.

**Comment by ghost** [comment]
Looks much better.  One very minor nit is that the button edges don't have transparency so the the white edges are visible on the blue rows' background, particularly the square corners.

**Comment by hflynn** [comment]
The transparency issue is really pretty minor, at least they all match now.
I'm happy to leave this here, looks fine to me!

**Comment by ghost** [comment]
It's certainly not worth spending time on fixing now, it's all fine with me.

**Comment by sbesson** [comment]
Thanks all. Next set of changes: OMERO.cpp.

**Comment by ghost** [comment]
@sbesson One error I just noticed was that the C++ build link should have been http://www.openmicroscopy.org/site/support/bio-formats5.1/developers/cpp.html

**Pull request merged**